sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -2348,11 +2348,56 @@ def _import_zip_as_action_package(
                 log.exception(msg)
                 return ActionResult.make_failure(msg).as_dict()
 
+    def m_drop_data_source(
+        self,
+        datasource: DatasourceInfoTypedDict,
+        data_server_info: DataServerConfigTypedDict,
+    ) -> ActionResultDict:
+        from sema4ai_code.data.data_server_connection import DataServerConnection
+
+        http = data_server_info["api"]["http"]
+        auth = data_server_info["auth"]
+        user = auth["username"]
+        password = auth["password"]
+
+        connection = DataServerConnection(
+            http_url=f"http://{http['host']}:{http['port']}",
+            http_user=user,
+            http_password=password,
+        )
+
+        name = datasource.get("name")
+        engine = datasource["engine"]
+        created_table = datasource.get("created_table")
+        model_name = datasource.get("model_name")
+
+        query = ""
+        if engine == "files":
+            query = f"DROP TABLE files.{created_table}dsada"
+
+        elif name == "custom" and created_table:
+            query = f"DROP TABLE {name}.{created_table}"
+
+        elif engine.startswith("prediction:") or (name == "custom" and model_name):
+            query = f"DROP MODEL IF EXISTS {name}.{datasource.get('model_name')}"
+        else:
+            query = f"DROP DATABASE IF EXISTS {name}"
+
+        try:
+            connection.run_sql(query)
+        except Exception as e:
+            log.exception(f"Error while executing query: {query}. {str(e)}")
+            return ActionResult.make_failure(
+                f"Error while dropping data source."
```

Comment:
Should probably add the error when dropping the data source.

i.e.: `f"Error while dropping data source: {e}"`